src/wallet/test/wallet_tests.cpp

```diff
@@ -867,6 +867,16 @@ BOOST_FIXTURE_TEST_CASE(ZapSelectTx, TestChain100Setup)
     TestUnloadWallet(std::move(wallet));
 }
 
+class FailCursor : public DatabaseCursor
+{
+private:
+    bool m_pass{true};
+
+public:
+    explicit FailCursor(bool pass) : m_pass(pass) {}
+    bool Next(CDataStream& key, CDataStream& value, bool& complete) override { return m_pass; }
```

Comment:
nit in f91be084:

I'm not sure that this line make much sense. Return true without setting any key, value, nor the complete flag sounds like something that could cause a bug in the future. Might be better to just return false for now.
(same for the `DummyCursor` class)

Reply:
Done